common-pipeline-steps/src/main/scala/com/acxiom/pipeline/steps/MappingSteps.scala

```diff
@@ -0,0 +1,239 @@
+      }
+    })
+
+    // select expressions from dataframe and return
+    dataFrame.select(finalExprs: _*)
+  }
+
+  /**
+    * reorder columns in a dataframe to match order of a destination schema optionally adding new columns to the end
+    * @param dataFrame          the input dataframe to be reordered
+    * @param destinationSchema  the destination schema driving the column order
+    * @param addNewColumns      a flag determining whether missing attributes should be added
+    * @return   a new dataframe with column in order specified by destination schema
+    */
+  def orderAttributesToDestinationSchema(dataFrame: DataFrame, destinationSchema: StructType, addNewColumns: Boolean = true): DataFrame = {
+    // generate expressions that pull all destination attributes in order
+    val destExprs = destinationSchema.map(a => col(a.name))
+
+    val finalExprs = if(addNewColumns) {
+      // add expressions for any columns in the dataframe that are not in the destination schema
+      destExprs ++ dataFrame.columns.flatMap(x => {
+        if(!destinationSchema.exists(_.name == x)) {
+          logger.info(s"appending column '$x' to the end of destination schema")
+          List(col(x))
+        } else { List() }
+      })
+    } else { destExprs }
+
+    // select expressions from dataframe and return
+    dataFrame.select(finalExprs: _*)
+  }
+
+  /**
+    * renames columns based on inputAliases provided in the mappings object
+    * @param dataFrame  the input dataframe
+    * @param mappings   the mappings object containing the inputAliases
+    * @return  a dataframe with updated column names
+    */
+  def applyAliasesToInputDataFrame(dataFrame: DataFrame, mappings: Mappings): DataFrame = {
+    // create a map of all aliases to the output name
+    val inputAliasMap = mappings.details.flatMap(m => {
+        m.inputAliases.map(_ -> m.outputField)
+    }).toMap
+
+    // create expression with aliases applied
+    val finalExprs = dataFrame.columns.map(c => {
+      if(inputAliasMap.get(c).nonEmpty && inputAliasMap(c) != c) {
+        logger.info(s"mapping input column '$c' to destination column '${inputAliasMap(c)}'")
+      }
+      col(c).as(inputAliasMap.getOrElse(c, c))
+    })
+
+    // select expression from dataFrame and return
+    dataFrame.select(finalExprs: _*)
+  }
+}
+
+
+case class MappingDetails(outputField: String, inputAliases: List[String] = List(), transform: Option[String] = None)
+case class Mappings(details: List[MappingDetails])
+
+case class Attribute(name: String, dataType: String) {
+  def toStructField: StructField = {
+    val dataType = this.dataType.toLowerCase match {
+      case "string" => DataTypes.StringType
+      case "double" => DataTypes.DoubleType
+      case "integer" => DataTypes.IntegerType
+      case "timestamp" => DataTypes.TimestampType
+      case _ => DataTypes.StringType
+    }
+    StructField(this.name, dataType)
+  }
+}
+case class Schema(attributes: Seq[Attribute])
```

Comment:
Should this be List instead of Seq for consistency?

Reply:
I originally started with a List, but since the order is important, I changed it to Seq.  From a processing standpoint, I don't think it matters since List in scala is an implementation of Seq.  My preference for Seq is cosmetic as it tells developers that order matters for this object.